### DISCUSSION TITLE
Automatically adjusting mechanism for latency

### DISCUSSION THREAD

**Discussion by laurfb** [Haivision/srt#2919] [General]
Hi.
Is there any mechanism or intention to implement a mechanism to automatically adjust, on the fly, the latency value?
In cases of network congestion, it may be very useful to automatically adjust/increase the latency value sequentially, in steps of 50 or 100 ms, compared to the preset or initial value, to maintain the stream reliability.
Based on RTT monitoring and the number of transmission errors upon crossing some RTT and Latency thresholds?
After which, after the disappearance of the congestion, when the values for RTT/number of transmision errors decrease and remain stable for a specific amount of time, to revert to the initial latency value?
regards,
laur

**Comment by maxsharabayko** [comment]
This contradicts a bit with a concept of fixed end-to-end latency in SRT.
An application can reconnect with a higher latency if it detects congestion. In case of a congestion packet are anyway lost, so loosing 2xRTT to reconnect might be not a big deal in low latency networks.
Implementing this logic in SRT itself feels like an overkill for now. But let's see, maybe this discussion raises a lot of "thumbs ups".
